app/lib/cds_importer/entity_mapper/base_mapper.rb

```diff
@@ -108,26 +108,22 @@ def missing_entity_filter_for(missing_entities)
           end
         end
 
-        def relative_primary_key_paths_for_secondary_node
-          primary_key_paths_for_secondary_node.each_with_object({}) do |path, acc|
-            relative_path = path.sub("#{mapping_path}.", '')
-
-            acc[relative_path] = primary_key_mapping[path]
-          end
+        def entity_composite_primary_key
+          Array.wrap(entity.primary_key)
         end
 
-        def primary_key_paths_for_primary_node
-          primary_node_paths = primary_key_mapping.keys - primary_key_paths_for_secondary_node
+        def database_entities_for(primary_model_instance)
+          filter = filter_for(primary_model_instance)
 
-          primary_node_paths.index_with { |path| primary_key_mapping[path] }
+          entity.where(filter).pluck(*entity.primary_key).map do |composite_primary_key|
```

Comment:
I wonder why we need to wrap after pluck 🤔

Reply:
When the primary key is singular (e.g. not a composite key), pluck returns an array with a depth of 1:

```ruby
[:foo, :bar]
```

When the primary key is a composite, pluck returns an array with a depth of 2:

```ruby
[[:foo, :baz], [:bar, :qux]]
```
I wrap so can iterate regardless of what the pluck method returns